python/CMakeLists.txt

```diff
@@ -0,0 +1,99 @@
+if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
+  # pybind11 logic for setting up a debug build when both a debug and release
+  # python interpreter are present in the system seems to be pretty much broken.
+  # This works around the issue.
+  set(PYTHON_LIBRARIES "${PYTHON_DEBUG_LIBRARIES}")
+endif()
+
+
+if(USE_SYSTEM_PATHS_FOR_PYTHON_INSTALLATION)
+  if(${CMAKE_VERSION} VERSION_LESS "3.12.0")
+    execute_process(
+      COMMAND "${PYTHON_EXECUTABLE}" -c "if True:
+  from distutils import sysconfig as sc
+  print(sc.get_python_lib(plat_specific=True))"
+      OUTPUT_VARIABLE Python3_SITEARCH
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+  else()
+    # Get install variable from Python3 module
+    # Python3_SITEARCH is available from 3.12 on, workaround if needed:
+    find_package(Python3 COMPONENTS Interpreter)
+  endif()
+
+  if(USE_DIST_PACKAGES_FOR_PYTHON)
+    string(REPLACE "site-packages" "dist-packages" IGN_PYTHON_INSTALL_PATH ${Python3_SITEARCH})
+  else()
+    # custom cmake command is returning dist-packages
+    string(REPLACE "dist-packages" "site-packages" IGN_PYTHON_INSTALL_PATH ${Python3_SITEARCH})
+  endif()
+else()
+  # If not a system installation, respect local paths
+  set(IGN_PYTHON_INSTALL_PATH ${IGN_LIB_INSTALL_DIR}/python)
+endif()
+
+set(IGN_PYTHON_INSTALL_PATH "${IGN_PYTHON_INSTALL_PATH}/ignition")
+
+set(PYBIND11_PYTHON_VERSION 3)
+
+find_package(pybind11 2.2 QUIET)
+
+if (NOT ${pybind11_FOUND})
```

Comment:
Let's only support `pybind11` from the system like we're doing on `ign-math`, otherwise it's tough to maintain multiple versions.